vhost_user_fs/src/descriptor_utils.rs

```diff
@@ -191,7 +194,14 @@ impl<'a> Reader<'a> {
     /// Construct a new Reader wrapper over `desc_chain`.
     pub fn new(mem: &'a GuestMemoryMmap, desc_chain: DescriptorChain<'a>) -> Result<Reader<'a>> {
         let mut total_len: usize = 0;
-        let buffers = desc_chain
+        let chain = if desc_chain.is_indirect() {
+            desc_chain
+                .new_from_indirect()
+                .map_err(|err| Error::ConvertIndirectDescriptor(err))?
```

Comment:
Clippy doesn't like this. Change to: `.map_err(Error::ConvertIndirectDescriptor)?`

Reply:
Fixed, thanks!